lib/core/test/unit/Cardano/Wallet/ApiSpec.hs

```diff
@@ -251,25 +257,32 @@ instance
         traverseLeft fn xs =
             fmap swap <$> traverse (traverse fn) (swap <$> xs)
 
+        insideOut :: [([x], y)] -> [(x, y)]
+        insideOut = foldl' (\y (xs, b) -> y ++ fmap (, b) xs) []
```

Comment:
**Suggestion**: Use ordinary list comprehension syntax.

**Justification**: Shorter and more readable. (Obviously this is subjective.)

```suggestion
        insideOut zs = [(x, y) | (xs, y) <- zs, x <- xs]
```

Reply:
I don't think that's even the same code. `foldl'` is strict in the accumulator. List comprehension desugars to something monadic, which I believe uses `concat` in the end, which uses `foldr`. So that may be optimised away due to list fusion... or not.

I also believe there should be some freedom to choose which constructs to use. I don't find list comprehension more readable.